protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -520,33 +530,93 @@ impl BridgeContractCounterparty for MovementClient {
+		let bridge_transfer_id_hex = format!("0x{}", hex::encode(bridge_transfer_id.0));
+	    
+		// Construct the ViewRequest
+		let view_request = ViewRequest {
+		    function: EntryFunctionId {
+			module: MoveModuleId {
+			    address: self.counterparty_address.clone().into(),
+			    name: aptos_api_types::IdentifierWrapper(Identifier::new("atomic_bridge_counterparty")
+				.map_err(|_| BridgeContractCounterpartyError::FunctionViewError)?),
+			},
+			name: aptos_api_types::IdentifierWrapper(Identifier::new("bridge_transfers")
+			    .map_err(|_| BridgeContractCounterpartyError::FunctionViewError)?),
+		    },
+		    type_arguments: vec![],
+		    arguments: vec![serde_json::json!(bridge_transfer_id_hex)],
+		};
+	    
+		// Send the request to the "/view" endpoint using JSON
+		let response: Response<Vec<serde_json::Value>> = self.rest_client
+		    .view(&view_request, None)
+		    .await
+		    .map_err(|_| BridgeContractCounterpartyError::CallError)?;
+	    
+		// Extract and parse the response
+		let values = response.inner();
+		if values.len() == 6 {
+		    let originator = values[0].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?;
+		    let recipient = values[1].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?;
+		    let amount = values[2].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?.parse::<u64>().map_err(|_| BridgeContractCounterpartyError::SerializationError)?;
+		    let hash_lock = values[3].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?;
+		    let time_lock = values[4].as_str().ok_or(BridgeContractCounterpartyError::SerializationError)?.parse::<u64>().map_err(|_| BridgeContractCounterpartyError::SerializationError)?;
+		    let state = values[5].as_u64().ok_or(BridgeContractCounterpartyError::SerializationError)? as u8;
+	    
+		    // Convert the originator, recipient, and hash_lock
+		    let originator_address = AccountAddress::from_hex_literal(originator)
+			.map_err(|_| BridgeContractCounterpartyError::SerializationError)?;
+		    let recipient_address_bytes = hex::decode(&recipient[2..])
+			.map_err(|_| BridgeContractCounterpartyError::SerializationError)?;
+		    let hash_lock_array: [u8; 32] = hex::decode(&hash_lock[2..])
+			.map_err(|_| BridgeContractCounterpartyError::SerializationError)?
+			.try_into()
+			.map_err(|_| BridgeContractCounterpartyError::SerializationError)?;
+	    
+		    // Create the BridgeTransferDetails struct
+		    let details: BridgeTransferDetails<MovementAddress, [u8; 32]> = BridgeTransferDetails {
+			bridge_transfer_id,
+			initiator_address: InitiatorAddress(MovementAddress(originator_address)),
+			recipient_address: RecipientAddress(recipient_address_bytes),
+			amount: Amount(AssetType::Moveth(amount)),
+			hash_lock: HashLock(hash_lock_array),
+			time_lock: TimeLock(time_lock),
+			state,
+		    };
+	    
+		    return Ok(Some(details));
+		}
+	    
+		Ok(None)
```

Comment:
`if values.len() != 6` here I would return an Error here rather than the `Ok` variant of Result.